x/nftmint/keeper/class_test.go

```diff
@@ -0,0 +1,363 @@
+	suite.Require().Error(err)
+
+	// in case which contains the invalid token supply cap
+	testMsgCreateClassInvalidTokenSupplyCap := types.MsgCreateClass{
+		Sender:            sender.Bytes(),
+		Name:              testName,
+		BaseTokenUri:      testBaseTokenUri,
+		TokenSupplyCap:    10000000, // bigger than the token supply cap
+		MintingPermission: 0,
+	}
+	err = suite.app.NftmintKeeper.CreateClass(suite.ctx, classIdInInvalidCase, &testMsgCreateClassInvalidTokenSupplyCap)
+	suite.Require().Error(err)
+
+	// in case which contains the invalid minting permission
+	testMsgCreateClassInvalidMintingPermission := types.MsgCreateClass{
+		Sender:            sender.Bytes(),
+		Name:              testName,
+		BaseTokenUri:      testBaseTokenUri,
+		TokenSupplyCap:    10000,
+		MintingPermission: 10, // not allowed minting permission option
+	}
+	err = suite.app.NftmintKeeper.CreateClass(suite.ctx, classIdInInvalidCase, &testMsgCreateClassInvalidMintingPermission)
+	suite.Require().Error(err)
+}
+
+// test for the SendClass relating functions
+func (suite *KeeperTestSuite) TestSendClass() {
+	sender := sdk.AccAddress(ed25519.GenPrivKey().PubKey().Address().Bytes())
+	sender_seq, _ := suite.app.AccountKeeper.GetSequence(suite.ctx, sender)
+	classId := keeper.CreateClassId(sender_seq, sender)
+	_ = suite.CreateClass(suite.ctx, classId, sender)
+
+	recipient := sdk.AccAddress(ed25519.GenPrivKey().PubKey().Address().Bytes())
+
+	testMsgSendClass := types.MsgSendClass{
+		Sender:    sender.Bytes(),
+		ClassId:   classId,
+		Recipient: recipient.Bytes(),
+	}
+	err := suite.app.NftmintKeeper.SendClass(suite.ctx, &testMsgSendClass)
+	suite.Require().NoError(err)
+	// check if recipient address becomes new owner of class
+	classAttributes, exists := suite.app.NftmintKeeper.GetClassAttributes(suite.ctx, classId)
+	suite.Require().True(exists)
+	expectedOwner := recipient
+	suite.Require().Equal(expectedOwner, classAttributes.Owner.AccAddress())
+
+	// invalid sender of MsgSendclass
+	invalidSender := sdk.AccAddress(ed25519.GenPrivKey().PubKey().Address().Bytes())
+
+	testMsgSendClassInvalidSender := types.MsgSendClass{
+		Sender:    invalidSender.Bytes(), // not the owner of class
+		ClassId:   classId,
+		Recipient: recipient.Bytes(),
+	}
+	err = suite.app.NftmintKeeper.SendClass(suite.ctx, &testMsgSendClassInvalidSender)
+	suite.Require().Error(err)
+
+	// invalid class id specification
+	invalidClassId := "nonexistance"
+	testMsgCreateClassInvalidClassId := types.MsgSendClass{
+		Sender:    sender.Bytes(),
+		ClassId:   invalidClassId, // non-existant class
+		Recipient: recipient.Bytes(),
+	}
+	err = suite.app.NftmintKeeper.SendClass(suite.ctx, &testMsgCreateClassInvalidClassId)
+	suite.Require().Error(err)
+}
+
+// test for the UpdateTokenSupplyCap relating functions
+func (suite *KeeperTestSuite) TestUpdateTokenSupplyCap() {
+	sender := sdk.AccAddress(ed25519.GenPrivKey().PubKey().Address().Bytes())
```

Comment:
I think it would be nice to write table driven test for this test since it has same function call with different cases - rather than writing all the tests inline

Reply:
Thanks for the advice.

Reply:
I see that table driven test is written for `TestUpdateBaseTokenUri(`, it would be nice from next time, more table driven test would be going for all the subjects for adding more scenarios next time, it's not mandatory to write all the tests as table driven though.

Reply:
Got it.